src/rust/src/x509/sign.rs

```diff
@@ -351,74 +351,46 @@ pub(crate) fn sign_data<'p>(
     signature.extract()
 }
 
-fn py_hash_name_from_hash_type(hash_type: HashType) -> Option<&'static str> {
-    match hash_type {
-        HashType::None => None,
-        HashType::Sha224 => Some("SHA224"),
-        HashType::Sha256 => Some("SHA256"),
-        HashType::Sha384 => Some("SHA384"),
-        HashType::Sha512 => Some("SHA512"),
-        HashType::Sha3_224 => Some("SHA3_224"),
-        HashType::Sha3_256 => Some("SHA3_256"),
-        HashType::Sha3_384 => Some("SHA3_384"),
-        HashType::Sha3_512 => Some("SHA3_512"),
-    }
-}
-
-pub(crate) fn verify_signature_with_oid<'p>(
+pub(crate) fn verify_signature_with_signature_algorithm<'p>(
     py: pyo3::Python<'p>,
     issuer_public_key: &'p pyo3::PyAny,
     signature_algorithm: &common::AlgorithmIdentifier<'_>,
     signature: &[u8],
     data: &[u8],
 ) -> CryptographyResult<()> {
     let key_type = identify_public_key_type(py, issuer_public_key)?;
-    let (sig_key_type, sig_hash_type) =
-        identify_key_hash_type_for_algorithm_params(&signature_algorithm.params)?;
+    let sig_key_type = identify_key_type_for_algorithm_params(&signature_algorithm.params)?;
     if key_type != sig_key_type {
         return Err(CryptographyError::from(
             pyo3::exceptions::PyValueError::new_err(
                 "Signature algorithm does not match issuer key type",
             ),
         ));
     }
-    let sig_hash_name = py_hash_name_from_hash_type(sig_hash_type);
-    let hashes = py.import(pyo3::intern!(py, "cryptography.hazmat.primitives.hashes"))?;
-    let signature_hash = match sig_hash_name {
-        Some(data) => hashes.getattr(data)?.call0()?,
-        None => py.None().into_ref(py),
-    };
-
+    let py_signature_algorithm_parameters =
+        certificate::identify_signature_algorithm_parameters(py, signature_algorithm)?;
+    let py_signature_hash_algorithm =
+        certificate::identify_signature_hash_algorithm(py, signature_algorithm)?;
```

Comment:
I'm going to merge these as-is, but can you please do a follow-up PR to move them from `certificate.rs` to `sign.rs`.